src/solution.rs

```diff
@@ -0,0 +1,97 @@
+use std::process::Command;
+use anyhow::{anyhow, Result};
+use crate::Clash;
+use crate::outputstyle::OutputStyle;
+
+mod suite_run;
+use suite_run::SuiteRun;
+mod test_run;
+use test_run::TestRun;
+
+use self::test_run::TestRunResult;
+
+pub struct Solution {
+    clash: Clash,
+    build_command: String,
+    run_command: String,
+}
+
+impl Solution {
+    pub fn new(clash: Clash, build_command: String, run_command: String) -> Self {
+        Self { clash, build_command, run_command }
+    }
+
+    pub fn run(&self, ignore_failures: bool) -> Result<SuiteRun> {
+        self.build()?;
```

Comment:
I don't like that this always calls `self.build`, I would prefer decoupling the two and making `build` public so the caller has to explicitly decide when (and if) build needs to be called.

Reply:
This should be resolved.